lib/ridgepole/diff.rb

```diff
@@ -177,13 +195,16 @@ def scan_options_change(table_name, from, to, table_delta)
     end
 
     def convert_to_primary_key_attrs(column_options)
-      options = column_options.dup
+      type = Ridgepole::DSLParser::TableDefinition::DEFAULT_PRIMARY_KEY_TYPE
+      options = column_options.deep_dup
 
-      type = if options[:id]
-               options.delete(:id)
-             else
-               Ridgepole::DSLParser::TableDefinition::DEFAULT_PRIMARY_KEY_TYPE
-             end
+      if ActiveRecord.gem_version < Gem::Version.new('6.1.0')
+        type = options.delete(:id) if options[:id]
+      else
+        id = options[:id].is_a?(Hash) ? options.delete(:id) : {}
+        type = id.delete(:type) if id[:type]
+        options.merge!(id.slice(*PRIMARY_KEY_OPTIONS))
+      end
```

Comment:
ActiveRecord version checking makes the code hard to read, so I think it's better to eliminate it as much as possible.
How about modifying it as follows? 

```suggestion
      options_id = options.delete(:id)

      if options_id.is_a?(Hash)
        type = options_id.delete(:type) if options_id[:type]
        options.merge!(options_id.slice(*PRIMARY_KEY_OPTIONS))
      elsif options_id
        type = options_id
      end
```

Reply:
applied 😄 
NOTE: I removed `id_options = options.delete(:id)` at the first line to avoid deleting `options[:id]` when id is `false`.